includes/class-newspack-popups-inserter.php

```diff
@@ -76,6 +81,29 @@ public static function popup( $content = '' ) {
 		return $content;
 	}
 
+	/**
+	 * Process popup and insert into archive pages if needed. Applies to Newspack Theme only.
+	 */
+	public static function popup_after_header() {
+		/* Posts and pages are covered by the_content hook */
+		if ( is_user_logged_in() || is_single() || is_page() ) {
```

Comment:
Just a fun tip: [is_singular](https://developer.wordpress.org/reference/functions/is_singular/) is an easier way of doing `is_single() || is_page()`